Documentation/D2.md

```diff
@@ -21,17 +21,30 @@ During our first meeting we discovered that:
 
 ## Metrics:
 We plan on evaluating the following metrics in order:
-* Response Time (Time between local commit, remote push, or pull request and recorded output event from the RPi code)
+* Response Time
+  * Experiment: We will be measuring the response time of the device from the time the user preforms a valid action to the time the device is triggered to respond. Valid actions are: local commits, remote pushes, or pull requests. The time the device sends the signal to activate the connected devices.
+  * Procedure: We will be preforming a retroactive analysis of the events that occure. Step one is to preform 30 valid actions. We will then pull the github logs timestamps and the RPis logs timestamps and compute the average difference to determine action laitency.
+  * Expected Results: We expect the results to be less than two seconds from the time a user preforms a valid action to when the device responds.
   * Independent Variables:
     * Git is a web service that relies on https protocols and tcp/ip packets through dns servers and our times to connect to those services could be affected by internet volume, security checks, and availability of the services. While it may be possible to call the GitHub servers directly to bypass the dns lookup, this should not significantly effect the time to communicate with the services.
+    * Repository size and commit size could be a potential factor. When pull requests are accepted they are merged which in large repositories and commits could take longer to complete. Because we are not interested in the performance of githubs algorythms we will be limiting the number of lines needed to merge to no more than 20 line changes.
 * Cost and marketability (A survey with standard questions that contribute to the goal of obtaining the level of interest, user acceptance, and device pricing)
+  * Experiment: We will attempt to determine the end user marketability of our device by surveying a population of developers familure with github.
+  * Procedure: Using an anonymous written(digitial or paper) survey with questions pertaining to user acceptance of the device using a veriety of dichotomous and Likert scales. The Likert scales will be built using current standards according to reasearch using a 5 point scale with the polor lables being delight and frustration.
+  * Expected Results: We expect the results to revieal that users will be more motivated to preform valid actions, and find programming more enjoyable or whimsicle.
```

Comment:
* Expected Results: We expect the tests to give us sufficient evidence to conclude our hypothesis where the programming hat is a marketable incentive which helps developers program more and contribute to Github.